### PR TITLE
[android] Improved touch target size 

### PR DESCRIPTION
**Pull request by SigmaZardan**
This PR addresses the enhancement of touch target sizes for various UI elements in the app, optimizing the user experience on touch-enabled devices. The changes were identified using Android Studio's built-in feature, which highlighted UI elements in XML files requiring adjustments for better touch interaction.

Screenshots
Attached screenshots showcasing the changes are available in this PR.

Major UI Element Changes
1. Toolbar Search Control Textfield:
Increased touch target size to improve ease of use during search operations.

2. Button with Progress and Cancel Button:
Adjusted touch targets for better accessibility during progress and cancellation operations.

3. Map Download Button (First-time Open):
Enlarged touch area for the button displayed when downloading maps on initial application launch.

4. Share Button:
Enhanced touch target size for improved accessibility when sharing content.

5. Cancel Buttons:
Increased touch target sizes for cancel buttons across various dialogs.

6. Dialog Edit Text Field (Bookmark List Screen):
Adjusted touch targets for better usability in the bookmark list screen.

7. Textfields in Edit Bookmark:
Optimized touch target sizes for textfields in the edit bookmark functionality.

Testing
The changes have been thoroughly tested on the Android emulator (Pixel 5, API level 34) to ensure compatibility and a seamless user experience.

Screenshots
Screenshot 2023-12-04 at 13 11 01
Screenshot 2023-12-04 at 13 12 39
Screenshot 2023-12-04 at 13 14 52
Screenshot 2023-12-04 at 13 16 14
Screenshot 2023-12-04 at 13 17 22
Screenshot 2023-12-04 at 13 19 12
Screenshot 2023-12-04 at 13 23 58
Screenshot 2023-12-04 at 13 25 24
Screenshot 2023-12-04 at 13 34 45

### CODE REVIEW

**Comment by biodranik** [comment]
Thanks for the very important improvement! Can you please make the whole My Places text area clickable, not only the "..more" button?

**Comment by Jean-BaptisteC** [comment]
Thanks for your changes, be careful to do not break UI on little devices. Increase touch area is useful when we have other clickable elements around and user can missclick on other buttons.
In my opinion increase size of button on downloader layout is not necessary because we do not have other button near around.
Maybe follow Material guidelines to improve this.

**Comment by SigmaZardan** [comment]
@biodranik 

Yeah sure. I will try that and will also look for the DCO action.

**Comment by SigmaZardan** [comment]
@Jean-BaptisteC 

Thank you for clarifying that to me. I will surely look into the Material guidelines. Are there any specific devices that you want me to test on ?

**Comment by Jean-BaptisteC** [comment]
You can test with Nexus 5 (most little device on Android Studio).

**Comment by SigmaZardan** [comment]
@biodranik When I try to add a new place information we sure can add a title and a description but I don't see a section to add an author. The author is still present below the title in the UI but there is no input for that field while adding. I can add click listener to the title but if we tap below the title then it will do nothing. So, there are two options that we can approach : 
1. We can remove the author and then add click listener to the title making the entire section to adapt to the feature for expansion.
2. If the author is something to be added in the future then we can keep it as it is and add click listener to the entire elements. 
Can you please guide me on this one ?
Screenshot 2023-12-04 at 17 28 05

**Comment by SigmaZardan** [comment]
@Jean-BaptisteC  I have read the material design guidelines. Well, the guidelines do suggest about making touch targets at least 48dp * 48dp,other than that I could not find any other related information. But I did test on Nexus 5 and the UI looks and works as expected. And for the download button, I did the changes based on android studio's guidelines earlier and it is working fine. Shall I undo or keep the change for the button ?  
Screenshot 2023-12-04 at 18 07 10

**Comment by Jean-BaptisteC** [comment]
Thanks for check,
I prefer default material button size with elevation, I don't like big buttons. And I consider if users think elements of the interface is too tiny, it's possible to increase size of elements in Android settings.

**Comment by SigmaZardan** [comment]
@biodranik sure

**Comment by SigmaZardan** [comment]
@biodranik  I have made the changes and here is the result.

https://github.com/organicmaps/organicmaps/assets/94971168/1c6144d3-c19d-488c-98e3-19d5b0b83bc0

**Comment by SigmaZardan** [comment]
@Jean-BaptisteC Thats a valid point. I will go with the default button size for the download button.

**Comment by Jean-BaptisteC** [comment]
Maybe hide more button, when we have expand description ?

**Comment by SigmaZardan** [comment]
@Jean-BaptisteC Sure I will look into it

**Comment by SigmaZardan** [comment]
@Jean-BaptisteC  I have also undone the UI modifications made to the download button.

File: android/app/src/main/res/layout/button_with_progress.xml [hunk 1]
```diff
@@ -32,6 +32,7 @@
     android:layout_marginStart="@dimen/margin_base_plus"
     android:layout_marginEnd="@dimen/margin_base_plus"
     android:layout_marginBottom="@dimen/margin_base_plus"
+      android:padding="@dimen/margin_half"
```
Comment (Jean-BaptisteC) [inline android/app/src/main/res/layout/button_with_progress.xml:35]:
```suggestion
    android:padding="@dimen/margin_half"
```

File: android/app/src/main/res/layout/dialog_edit_text.xml [hunk 2]
```diff
@@ -25,6 +25,7 @@
       android:layout_height="wrap_content"
       android:inputType="text|textCapSentences"
       android:maxLength="100"
+        android:padding="@dimen/margin_half_double_plus"
```
Comment (Jean-BaptisteC) [inline android/app/src/main/res/layout/dialog_edit_text.xml:28]:
```suggestion
      android:padding="@dimen/margin_half_double_plus"
```

File: android/app/src/main/res/layout/item_cuisine.xml [hunk 3]
```diff
@@ -1,25 +1,24 @@
...
+    android:gravity="center_vertical"
+    android:orientation="horizontal"
+    android:padding="@dimen/margin_base">
 
-  <CheckBox
-    android:id="@+id/selected"
-    android:layout_width="wrap_content"
-    android:layout_height="wrap_content"
-    android:layout_marginEnd="@dimen/margin_double"/>
+    <CheckBox
```
Comment (Jean-BaptisteC) [inline android/app/src/main/res/layout/item_cuisine.xml:10]: Please fix indent

File: android/app/src/main/res/layout/item_street.xml [hunk 4]
```diff
@@ -1,45 +1,44 @@
...
-  xmlns:android="http://schemas.android.com/apk/res/android"
-  xmlns:tools="http://schemas.android.com/tools"
-  android:layout_width="match_parent"
-  android:layout_height="@dimen/height_item_oneline"
-  android:background="?clickableBackground"
-  android:gravity="center_vertical"
-  android:orientation="horizontal"
-  android:paddingEnd="@dimen/margin_base"
-  android:paddingStart="@dimen/margin_half_plus">
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (Jean-BaptisteC) [inline android/app/src/main/res/layout/item_street.xml:2]: Indent

**Comment by SigmaZardan** [comment]
@Jean-BaptisteC Sure I will do that

**Comment by SigmaZardan** [comment]
@Jean-BaptisteC  I did the changes only with two commits as you mentioned. Is it fine ?

**Review by Jean-BaptisteC — APPROVED** [review]
LGTM

Please remove changes on size button to download map and rebase branch

**Comment by SigmaZardan** [comment]
@Jean-BaptisteC Could you please clarify if you are referring to this button ? I want to make sure I understand your query correctly.

**Comment by Jean-BaptisteC** [comment]
Not download button on second screenshot

**Comment by SigmaZardan** [comment]
@Jean-BaptisteC sure i will do that

**Comment by SigmaZardan** [comment]
@Jean-BaptisteC I have done the required changes

**Comment by Jean-BaptisteC** [comment]
Please remove your merge branch commits.

**Comment by biodranik** [comment]
We can try to squash commits without rebasing, if everything else is ok. @Jean-BaptisteC are all proposed changes working ok?

**Review by Jean-BaptisteC — APPROVED** [review]
LGTM
I have tested changes, that doesn't break UI

**Review by biodranik — APPROVED** [review]
Thank you!

**Pull request merged**